### ISSUE TITLE
Error creating embedded templates when there are no merge fields

### ISSUE THREAD

**Issue by codesometech** [HelloFax/hellosign-java-sdk#55]
I am seeing the below exception while creating an embedded template using the Java SDK 4.0.3. I see that the request object has "[]" for merge_fields. I this it is because of that. Can you please take a look. I am including screen shots and code snippet from TemplateDraft.java for your reference. **I think an empty check on mergeFileds map should fix the issue.**

```
com.hellosign.sdk.HelloSignException: Merge field is missing "name" parameter
	at com.hellosign.sdk.http.HttpClient.validate(HttpClient.java:177)
	at com.hellosign.sdk.http.HttpClient.asJson(HttpClient.java:210)
	at com.hellosign.sdk.HelloSignClient.createEmbeddedTemplateDraft(HelloSignClient.java:939)
```
![image](https://cloud.githubusercontent.com/assets/3298948/25027105/5238560c-2078-11e7-8849-4802ac5e3168.png)

![image](https://cloud.githubusercontent.com/assets/3298948/25027093/3f9544b0-2078-11e7-976f-e1f0c6092745.png)

**Comment by mveitas** [comment]
PRs are always welcome, hence why this is open source :)

**Comment by cmpaul** [comment]
@codesometech Thanks for the screenshots and explanation, seems straightforward enough. I'll see if I can get a patch out in the next few days.

**Comment by cmpaul** [comment]
@codesometech Can you take a look at PR #57 and let me know if that will work for you?

**Comment by cmpaul** [comment]
@codesometech This has been merged to develop. If you're so inclined, you can check out that branch and build (it will build version 4.0.4-SNAPSHOT) to test. I'll plan to release the official patch bump next week.

**Comment by codesometech** [comment]
@cmpaul Thank you.

**Comment by codesometech** [comment]
@cmpaul I tested this locally and it looks good. Could you please release this.

**Comment by cmpaul** [comment]
@codesometech This has been released as v4.0.4. Please update and let me know if you have any issues. Thanks for your help!
